akka-stream/src/main/scala/akka/stream/impl/fusing/Interpreter.scala

```diff
@@ -398,8 +399,24 @@ private[akka] class OneBoundedInterpreter(ops: Seq[Stage[_, _]], val forkLimit:
         state.progress()
       } catch {
         case NonFatal(e) if lastOpFailing != activeOpIndex ⇒
+          if (Debug) printDebug()
           lastOpFailing = activeOpIndex
-          state.fail(e)
+          currentOp.decide(e) match {
+            case Supervision.Stop ⇒
+              println(s"# fail [$activeOpIndex]: $e") // FIXME
+              state.fail(e)
+            case Supervision.Resume ⇒
+              println(s"# resume [$activeOpIndex]: $e") // FIXME
+              // FIXME do we have to keep track of if user called ctx.push and exception was thrown afterwards
+              // FIXME `pull` can also throw
+              state.pull()
```

Comment:
yes, or I was thinking on defining `resume` and `restart` on the state, and it can do the right thing